onnx2kerastl/converter.py

```diff
@@ -134,10 +137,15 @@ def onnx_to_keras(onnx_model, input_names, name_policy=None, verbose=True, chang
     embedding_weights_mapping = {}
     for node_index, node in enumerate(onnx_nodes):
         if node.op_type == 'If':
-            if layers[node.input[0]][0]:
-                replace_node = node.attribute[0].g.node
-            else:
-                replace_node = node.attribute[1].g.node
+            cond = layers[node.input[0]][0]
+            if not isinstance(cond, bool) and not isinstance(cond, tf.Tensor) and keras.backend.is_keras_tensor(cond):
+                # the condition in If is a KerasTensor and needs to be evlauated.
+                inpt_sample = [tf.ones(inpt.shape) for inpt in keras_inputs]
+                cond = keras.models.Model(keras_inputs, cond)(inpt_sample)
+                if cond:
+                    replace_node = node.attribute[0].g.node
+                else:
+                    replace_node = node.attribute[1].g.node
```

Comment:
Aren't we missing an else block similar to this one? for the case `cond` is bool or tf.Tensor?
```suggestion
                    replace_node = node.attribute[1].g.node
                else:
                    if all(cond):
                        replace_node = node.attribute[0].g.node
                    else:
                        replace_node = node.attribute[1].g.node
                    
```

Reply:
wrong identation - fixed.